### PR TITLE
Reuse metric names in TopMetricsAggregator

### PR DESCRIPTION
**Pull request by iverase**
I noticed in a heapdump that we are creating a new list of names for each new InternalTopMetrics created which seems wasteful as we can be creating thousand of those objects during an aggregation execution. This commit shares a unique instance between all InternalTopMetrics instances.

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachine update branch

**Review by ivancea — APPROVED** [review]
LGTM

File: x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java [hunk 1]
```diff
@@ -148,9 +146,14 @@ public void doClose() {
 
     static class Metrics implements BucketedSort.ExtraData, Releasable {
         private final MetricValues[] values;
+        private final List<String> names;
 
         Metrics(MetricValues[] values) {
             this.values = values;
+            names = new ArrayList<>(values.length);
+            for (MetricValues value : values) {
```
Comment (not-napoleon) [inline x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java:154]: I agree that the for loop is more readable here than the stream.

**Review by not-napoleon — APPROVED** [review]
LGTM.

File: x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java (hunk 1 above)
Comment (iverase) [reply x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregator.java:154]: And the stream was creating a list with size 10  although we were adding only one element.

**Pull request merged**